src/Microsoft.ML.Core/Environment/HostEnvironmentBase.cs

```diff
@@ -125,7 +127,8 @@ public void StopExecution()
                     foreach (var child in _children)
                     {
                         if (child.TryGetTarget(out IHost host))
-                            host.StopExecution();
+                            if (host is ICancellableHost)
```

Comment:
use pattern matching here:

```c#
if (host is ICancellableHost cancellableHost)
{
    cancellableHost.StopExecution();
}
```

Reply:
and below.

---
In reply to: [264859870](https://github.com/dotnet/machinelearning/pull/2797#discussion_r264859870) [](ancestors = 264859870)